packages/fxa-settings/src/components/App/index.tsx

```diff
@@ -32,7 +37,42 @@ import SigninReported from '../../pages/Signin/SigninReported';
 export const App = ({
   flowQueryParams,
 }: { flowQueryParams: QueryParams } & RouteComponentProps) => {
+  const [isSignedIn, setIsSignedIn] = useState<boolean>();
+
   const { showReactApp } = flowQueryParams;
+  const { loading, error } = useInitialState();
+  const { metricsEnabled } = useAccount();
+  const config = useConfig();
+
+  useEffect(() => {
+    Metrics.init(metricsEnabled, flowQueryParams);
+  }, [metricsEnabled, flowQueryParams]);
+
+  useEffect(() => {
+    if (!loading && error?.message.includes('Invalid token')) {
```

Comment:
Kind of wonder if we could use a little helper function `isSignedIn` that takes in `error` and returns a boolean for this for clarity since we're using it in at least a couple of places now.